### PR TITLE
new package: GunRock

### PR DESCRIPTION
**Pull request by Sinan81**
High-Performance Graph Primitives on GPUs

### CODE REVIEW

File: var/spack/repos/builtin/packages/gunrock/package.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    version('develop', git=git, submodules=True)
+    version('1.1', git=git, submodules=True, tag='v1.1', preferred=True)
+    version('1.0', git=git, submodules=True, tag='v1.0')
+    version('0.5.1', git=git, submodules=True, tag='v0.5.1')
+    version('0.5', git=git, submodules=True, tag='v0.5')
+    version('0.4', git=git, submodules=True, tag='v0.4')
+    version('0.3.1', git=git, submodules=True, tag='v0.3.1')
+    version('0.3', git=git, submodules=True, tag='v0.3')
+    version('0.2', git=git, submodules=True, tag='v0.2')
+    version('0.1', git=git, submodules=True, tag='v0.1')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/gunrock/package.py]: Can we use tarballs for point releases?

Comment (Sinan81) [reply var/spack/repos/builtin/packages/gunrock/package.py]: Tarballs don't iclude the external git modules necessary to compile thile software

File: var/spack/repos/builtin/packages/gunrock/package.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+
+class Gunrock(CMakePackage, CudaPackage):
+    """High-Performance Graph Primitives on GPUs"""
+
+    homepage = "https://gunrock.github.io/docs/"
+    url      = "https://github.com/gunrock/gunrock/archive/v1.1.tar.gz"
+    git      = "https://github.com/gunrock/gunrock.git"
+
+    version('develop', git=git, submodules=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gunrock/package.py]:
```suggestion
    version('master', submodules=True)
```
If you use `submodules`, it must be a git download, and you have git specified at the top, so this should work.

File: var/spack/repos/builtin/packages/gunrock/package.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Gunrock(CMakePackage, CudaPackage):
+    """High-Performance Graph Primitives on GPUs"""
+
+    homepage = "https://gunrock.github.io/docs/"
+    url      = "https://github.com/gunrock/gunrock/archive/v1.1.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gunrock/package.py]: Just remove this URL if you aren't going to use it.

File: var/spack/repos/builtin/packages/gunrock/package.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    git      = "https://github.com/gunrock/gunrock.git"
+
+    version('develop', git=git, submodules=True)
+    version('1.1', git=git, submodules=True, tag='v1.1', preferred=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gunrock/package.py]: No need for `preferred=True` here, Spack will automatically choose the latest numbered release.

**Pull request merged**